gridstatus/ieso.py

```diff
@@ -0,0 +1,473 @@
+        ):
+            raise NotSupported(
+                f"Dates more than {MAXIMUM_DAYS_IN_FUTURE_FOR_LOAD_FORECAST}"
+                "days in the future are not supported for load forecasts.",
+            )
+
+        # For future dates, the most recent forecast is used
+        if date_only > today:
+            url = LOAD_FORECAST_TEMPLATE_URL.replace("_YYYYMMDD", "")
+        else:
+            url = LOAD_FORECAST_TEMPLATE_URL.replace(
+                "YYYYMMDD",
+                date.strftime("%Y%m%d"),
+            )
+
+        r = self._request(url, verbose)
+
+        # Define the NAMESPACES_FOR_XML used in the XML document
+        NAMESPACES_FOR_XML = {"": "http://www.ieso.ca/schema"}
+
+        # Initialize a list to store the parsed data
+        data = []
+
+        # Parse the XML file
+        root = ET.fromstring(r.content)
+
+        published_time = root.find(".//CreatedAt", NAMESPACES_FOR_XML).text
+
+        # Extracting data for each ZonalDemands within the Document
+        for zonal_demands in root.findall(".//ZonalDemands", NAMESPACES_FOR_XML):
+            delivery_date = zonal_demands.find(
+                ".//DeliveryDate",
+                NAMESPACES_FOR_XML,
+            ).text
+
+            for zonal_demand in zonal_demands.findall(
+                ".//ZonalDemand/*",
+                NAMESPACES_FOR_XML,
+            ):
+                # The zone name is the tag name without the namespace
+                zone_name = zonal_demand.tag[(zonal_demand.tag.rfind("}") + 1) :]
+
+                for demand in zonal_demand.findall(".//Demand", NAMESPACES_FOR_XML):
+                    hour = demand.find(".//DeliveryHour", NAMESPACES_FOR_XML).text
+                    energy_mw = demand.find(".//EnergyMW", NAMESPACES_FOR_XML).text
+
+                    data.append(
+                        {
+                            "DeliveryDate": delivery_date,
+                            "Zone": zone_name,
+                            "DeliveryHour": hour,
+                            "EnergyMW": energy_mw,
+                        },
+                    )
+
+        df = pd.DataFrame(data)
+
+        # Convert columns to appropriate data types
+        df["DeliveryHour"] = df["DeliveryHour"].astype(int)
+        df["EnergyMW"] = df["EnergyMW"].astype(float)
+        df["DeliveryDate"] = pd.to_datetime(df["DeliveryDate"])
+
+        df["Interval Start"] = (
+            df["DeliveryDate"] + pd.to_timedelta(df["DeliveryHour"], unit="h")
+        ).dt.tz_localize(self.default_timezone)
+
+        df["Interval End"] = df["Interval Start"] + pd.Timedelta(hours=HOUR_INTERVAL)
+
+        # Pivot the table to wide
+        pivot_df = df.pivot_table(
+            index=["Interval Start", "Interval End"],
+            columns="Zone",
+            values="EnergyMW",
+            aggfunc="first",
+        ).reset_index()
+
+        pivot_df["Published Time"] = pd.Timestamp(
```

Comment:
Let's name this column `Publish Time` to be consistent with ERCOT.

Reply:
Do you want a "Publish Time" column in the `get_load` data as well, or only the `get_load_forecast` data?

Reply:
Let’s do it for load forecast only since it is necessary there to figure out which load forecast vintage is being used.

that confusion doesn’t apply to load.